auditbeat/module/audit/file/event.go

```diff
@@ -23,71 +26,97 @@ func (a Action) String() string {
-	MovedTo:            "moved_to",
-	CollisionWithin:    "collision_within",
-	Unmounted:          "unmounted",
-	RootChanged:        "root_changed",
+	Moved:              "moved",
 }
 
+// Event describe the filesystem change and includes metadata about the file.
 type Event struct {
-	Path      string    `structs:"path"`   // The path associated with the event.
-	Action    string    `structs:"action"` // Action (like created, updated).
-	Inode     uint64    `structs:"inode"`
-	UID       int32     `structs:"uid"`
-	GID       int32     `structs:"gid"`
-	Owner     string    `structs:"owner,omitempty"`
-	Group     string    `structs:"group,omitempty"`
-	Mode      string    `structs:"mode"`
-	Size      int64     `structs:"size"`
-	ATime     time.Time `structs:"atime"` // Last access time.
-	MTime     time.Time `structs:"mtime"` // Last modification time.
-	CTime     time.Time `structs:"ctime"` // Last status change time.
-	MD5       string    `structs:"md5,omitempty"`
-	SHA1      string    `structs:"sha1,omitempty"`
-	SHA256    string    `structs:"sha256,omitempty"`
-	Hashed    bool      `structs:"hashed"`     // True if hashed.
-	Timestamp time.Time `structs:"@timestamp"` // Time of event.
+	Timestamp  time.Time // Time of event.
+	Path       string    // The path associated with the event.
+	TargetPath string    // Target path for symlinks.
+	Action     string    // Action (like created, updated).
+	Info       *Metadata // File metadata (if the file exists).
+	Hashed     bool      // True if hashed.
+	MD5        string
+	SHA1       string
+	SHA256     string
+
+	errors []error
+}
+
+// Metadata contains file metadata.
+type Metadata struct {
+	Inode uint64
+	UID   uint32
+	GID   uint32
+	SID   string
+	Owner string
+	Group string
+	Mode  os.FileMode
+	Size  int64
+	ATime time.Time // Last access time.
+	MTime time.Time // Last modification time.
+	CTime time.Time // Last status change time.
+	Type  string    // File type (dir, file, symlink).
 }
 
 func buildMapStr(e *Event) common.MapStr {
 	m := common.MapStr{
 		"@timestamp": e.Timestamp,
 		"path":       e.Path,
 		"action":     e.Action,
-		"inode":      e.Inode,
-		"uid":        e.UID,
-		"gid":        e.GID,
-		"mode":       e.Mode,
-		"size":       e.Size,
-		"atime":      e.ATime,
-		"mtime":      e.MTime,
-		"ctime":      e.CTime,
 		"hashed":     e.Hashed,
 	}
 
-	if e.Owner != "" {
-		m["owner"] = e.Owner
+	if e.TargetPath != "" {
+		m["target_path"] = e.TargetPath
 	}
-	if e.Group != "" {
-		m["group"] = e.Group
+
+	if e.Info != nil {
+		info := e.Info
+		m["inode"] = strconv.FormatUint(info.Inode, 10)
+		m["size"] = info.Size
+		m["atime"] = info.ATime
+		m["mtime"] = info.MTime
+		m["ctime"] = info.CTime
+
+		if info.Type != "" {
+			m["type"] = info.Type
+		}
+
+		if runtime.GOOS == "windows" {
+			if info.SID != "" {
+				m["sid"] = info.SID
+			}
+		} else {
+			m["uid"] = info.UID
+			m["gid"] = info.GID
+			m["mode"] = fmt.Sprintf("%#o", uint32(info.Mode))
```

Comment:
This will "add leading 0 for octal (%#o)". https://golang.org/pkg/fmt/

I was wanting something like `0644`, but maybe a different format string would be more appropriate because what I really want is a minimum of 4 octal digits.